messaging/src/main/java/org/axonframework/deadline/jobrunnr/JobRunrDeadlineManager.java

```diff
@@ -0,0 +1,298 @@
+        Span span = spanFactory.createDispatchSpan(() -> "JobRunrDeadlineManager.schedule(" + deadlineName + ")",
+                                                   deadlineMessage);
+        runOnPrepareCommitOrNow(span.wrapRunnable(() -> {
+            DeadlineMessage<Object> interceptedDeadlineMessage = processDispatchInterceptors(deadlineMessage);
+            DeadlineDetails deadlineDetails =
+                    new DeadlineDetails(deadlineName,
+                                        deadlineId,
+                                        deadlineScope,
+                                        interceptedDeadlineMessage.getPayload(),
+                                        new HashMap<>(interceptedDeadlineMessage.getMetaData()));
+            jobScheduler.<JobRunrDeadlineManager>schedule(deadlineId, triggerDateTime, x -> x.execute(deadlineDetails));
+        }));
+        return deadlineId.toString();
+    }
+
+    @Override
+    public void cancelSchedule(@Nonnull String deadlineName, @Nonnull String scheduleId) {
+        Span span = spanFactory.createInternalSpan(
+                () -> "JobRunrDeadlineManager.cancelSchedule(" + deadlineName + "," + scheduleId + ")");
+        runOnPrepareCommitOrNow(span.wrapRunnable(() -> jobScheduler.delete(toUuid(scheduleId),
+                                                                            "Deleted via DeadlineManager API")));
+    }
+
+    @Override
+    public void cancelAll(@Nonnull String deadlineName) {
+        throw new DeadlineException(
+                "The 'cancelAll' method is not implemented for JobRunrDeadlineManager, use 'cancelSchedule' instead.\n"
+                        + "This requires keeping track of the return value from 'schedule'.");
+    }
+
+    private UUID toUuid(@Nonnull String scheduleId) {
+        try {
+            return UUID.fromString(scheduleId);
+        } catch (IllegalArgumentException e) {
+            throw new DeadlineException("For jobrunr the scheduleId should be an UUID representation.", e);
+        }
+    }
+
+    @Override
+    public void cancelAllWithinScope(@Nonnull String deadlineName, @Nonnull ScopeDescriptor scope) {
+        throw new DeadlineException(
+                "The 'cancelAllWithinScope' method is not implemented for JobRunrDeadlineManager, use 'cancelSchedule' instead.\n"
+                        + "This requires keeping track of the return value from 'schedule'.");
+    }
+
+
+    /**
+     * This function should only be called via Jobrunr when a deadline was triggered. It will try to execute the
+     * scheduled deadline on the set scope. It will throw a {@link DeadlineException} in case of errors such that they
+     * will be optionally retried by Jobrunr.
+     *
+     * @param deadlineDetails the {@link DeadlineDetails} object with all the needed details to execute
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public void execute(@Nonnull DeadlineDetails deadlineDetails) {
```

Comment:
Does it need to be `public` for that to work? Does `protected` or package-private cut it, for example? Guess not actually, so just checking with you here.

Reply:
I tried it, but it's indeed not working, causing an illegal access exception.